angular-spa-ui/js/search-app/search-module/search-ctrl.js

```diff
@@ -1,58 +1,179 @@
+        //Change current searchIn
+        $scope.setSearchIn = function (val) {
+            $scope.searchIn = $scope.searchInList[findValue(val, $scope.searchInList)];
+            $scope.queryParams.searchIn = $scope.searchIn.value;
+            if ($scope.hasQuery()){
+                $state.go(
+                    'search.simpleQuery',
+                    $scope.queryParams,
+                    {
+                        reload : true
+                    }
+                );
+            }
+            
+        };
+        
+        // Function to find smth
+        $scope.find = function () {
+            if ($scope.hasQuery()){
+                $scope.queryParams.query = $scope.query;
+                $scope.showResults = false;
+                $state.go(
+                    'search.simpleQuery',
+                    $scope.queryParams,
+                    {
+                        inherit : true,
+                        reload : true
+                    }
+                );
+            }
+            
+        };
+        
+        //Action when sortBy changed
+        $scope.sortChange = function () {
+            $scope.queryParams.sortBy = $scope.sortBy.value;
+            $state.go(
+                'search.simpleQuery',
+                $scope.queryParams,
+                {
+                    reload : true 
+                }
+            );
+        };
+        
+        //Action when limit changed
+        $scope.limitChange = function () {
+            $scope.queryParams.limit = $scope.limit.value;
+            $state.go(
+                'search.simpleQuery',
+                $scope.queryParams,
+                {
+                    reload : true
+                }
+            );
         };
-    });
-    
-    search.controller('ModalDemoCtrl', function ($scope, $uibModal, $log) {
-         
-        $scope.items = ['item1', 'item2', 'item3'];
-
-        $scope.animationsEnabled = true;
-
-        $scope.open = function (size) {
 
-        var modalInstance = $uibModal.open({
-              animation: $scope.animationsEnabled,
-              templateUrl: 'myModalContent.html',
-              controller: 'ModalInstanceCtrl',
-              size: size,
-              resolve: {
-                items: function () {
-                  return $scope.items;
+        //Pagination change page
+        $scope.goToPage = function () {
+            $scope.queryParams.offset = $scope.currentPage * $scope.limit.value - $scope.limit.value;
+            $state.go(
+                'search.simpleQuery',
+                $scope.queryParams,
+                {
+                    inherit : true,
+                    reload : true
+                }
+            );
+        };
+        
+        // Check has model "query" data or not - if yes returns true, else false
+        $scope.hasQuery = function () {
+            if ($scope.query) {
+                return true;
+            } else {
+                return false;
```

Comment:
You do not need 'else' statement this:

if (...) {
return true;
}
return false;